runtime/sema/type.go

```diff
@@ -1933,6 +1996,10 @@ func (*ConstantSizedType) IsType() {}
 
 func (*ConstantSizedType) isArrayType() {}
 
+func (t *ConstantSizedType) Tag() TypeTag {
+	return ArrayTypeTag
```

Comment:
Type tag is only used as a fail-fast/early-exit method for more straightforward types. But for more complex types, we need to do more complex analysis (done in `getSuperTypeOfDerivedTypes`), and cannot only depend on the bitmask. So here we use one tag for both types, only to determine that it's an array type and would need more complex analysis.

I think its no harm in splitting/introducing two tags, but at this point, I think it would not add any additional value, as both would end up going on the same execution path. We could re-visit it though when we do the recursive element supertype inferring.

Reply:
Added a test case here to make sure the output is correct: https://github.com/onflow/cadence/pull/1025/commits/063795b564058e5c97ce60814ecc1917bd26da3d

Reply:
actually, on second thought, we might need it soon. So added it in https://github.com/onflow/cadence/pull/1025/commits/31abd0a4d37f2f82556737baa6cd0c8cac0171ab